### PR TITLE
fix: debounce bug

### PR DESCRIPTION
**Pull request by ArkadiK94**
**This pull request fixes the debounce bug**
- related to issue #38 
@ShayAdler @NoamGaash

### CODE REVIEW

File: src/pages/components/LineSelector.tsx [hunk 1]
```diff
@@ -10,17 +10,23 @@ type LineSelectorProps = {
 }
 
 const LineSelector = ({ lineNumber, setLineNumber }: LineSelectorProps) => {
-  const debouncedSetLineNumber = useCallback(debounce(setLineNumber), [setLineNumber])
+  const [value, setValue] = useState<LineSelectorProps['lineNumber']>()
+  const debouncedSetLineNumber = useCallback(debounce(setLineNumber, 200), [setLineNumber])
 
-  const value = lineNumber ? lineNumber : ''
+  useLayoutEffect(() => {
```
Comment (NoamGaash) [inline src/pages/components/LineSelector.tsx:16]: Why did you choose `useLayoutEffect` over `useEffect`?

**Review by NoamGaash — COMMENTED** [review]
Great job! Thank you for your help,
I just have a little question regarding an implementation detail

**Comment by ArkadiK94** [comment]
Hi, about the reason why I used useLayoutEffect. First, I tried with useEffect but when I chose the details of the bus number and moved to another tab in the navigation (where this component is also used) , I saw the change of this input filed from the place holder to the lineNumber data. So, I decided to useLayoutEffect which fires before the browser repaints the screen which helps to render the initial lineNumber content. However,  I can create a Loading instead,  if you prefer.

**Comment by ArkadiK94** [comment]
@NoamGaash Pls, Tell me if you prefer that I will change to useEffect with Loading instead of using useLayoutEffect.

**Comment by NoamGaash** [comment]
@ShayAdler @ArkadiK94 
I'm not sure why the test isn't passing.
I've tried fixing and enhancing the tests on this branch -
https://github.com/hasadna/open-bus-map-search/pull/45
I'll try to fix it there.

**Review by NoamGaash — APPROVED** [review]
I'm unhappy regarding the `useLayoutEffect` hook - we should find a better approach to store and share input data across pages.
but that's out of the scope of this PR - it's cool and it works. Let's merge it despite the growing technical debt.
thank you for your contribution! it's great :clap:

**Pull request merged**